integr8ly/integreatly-operator
# MGDAPI-4174 Configure mkdocs and readthedocs for the integreatly operator project - Restructure Documentation

# Issue link
[MGDAPI-4174](https://issues.redhat.com/browse/MGDAPI-4174)

# What
 - [MkDocs](https://www.mkdocs.org/) has been configured on this branch of the integreatly-operator repo and a [readthedocs](https://readthedocs.org/) site has been published [here](https://integreatly-operator.readthedocs.io/en/latest/). 
This site is automatically updated on PR merges.
 - The top level readme has been moved to the docs folder and named index.md as required for configuration.
 - The original readme has been broken down into smaller pieces and added to the docs folder (and are now displayed on the readthedocs site).
 - A lightweight introductory readme has been created in its place.
 - [RHOAM Team Permissions](https://docs.google.com/document/d/1LA-VaYZge1GYPIWo4XRHAW9DMrHyIeD2DNV69PovCcQ/edit) document has been updated to include Read the Docs.
 - A new make target has been added allowing the serving of the documentation site while doing local development. `mkdocs/serve`

NOTE: Some links in documents inside Docs folder are broken and need to be updated. A follow on [Jira ](https://issues.redhat.com/browse/MGDAPI-4393)has been created for this task.

# Verification steps
Verify that the new make target runs successfully
MkDocs requires a recent version of Python and python package manager pip to be installed on your system. 
 - Install mkdocs `pip install mkdocs`
 - Using this [PR](https://github.com/integr8ly/integreatly-operator/pull/2813) run the new make target. `make mkdocs/serve`
 - Navigate to `http://127.0.0.1:8000/` and you should see a local run version of the readthedocs site.
